src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019-2022 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.cleaner.archiveservice;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.function.Function;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.context.ManagedExecutor;
+import org.jboss.pnc.api.constants.MDCKeys;
+import org.jboss.pnc.cleaner.auth.KeycloakServiceClient;
+import org.jboss.pnc.common.Strings;
+import org.jboss.pnc.common.otel.OtelUtils;
+import org.slf4j.MDC;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Scope;
+
+import lombok.extern.slf4j.Slf4j;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+
+import static org.jboss.pnc.api.constants.HttpHeaders.AUTHORIZATION_STRING;
+import static org.jboss.pnc.api.constants.HttpHeaders.CONTENT_TYPE_STRING;
+
+/**
+ * Implementation of the ArchivesCleaner interface which provides methods to clean the historical archives stored in the
+ * Indy Archive Service
+ *
+ * @author Andrea Vibelli
+ */
+
+@ApplicationScoped
+@Slf4j
+public class ArchivesCleanerImpl implements ArchivesCleaner {
+
+    @Inject
+    KeycloakServiceClient serviceClient;
+
+    @Inject
+    Config config;
+
+    @Inject
+    ManagedExecutor executor;
+
+    @Inject
+    java.net.http.HttpClient httpClient;
+
+    private String DELETE_ARCHIVE_URL;
```

Comment:
Is there a reason for not using ConfigProperty from MP Config?

Reply:
no, it's some leftover from previous code which I haven't cleaned up. Fixing :-)